laws/src/main/scala/cats/laws/discipline/ExhaustiveCheck.scala

```diff
@@ -0,0 +1,55 @@
+package cats
+package laws
+package discipline
+
+/**
+ * An `ExhuastiveCheck[A]` instance can be used similarly to a Scalacheck
+ * `Gen[A]` instance, but differs in that it generates a `Stream` of the entire
+ * domain of values as opposed to generating a random sampling of values.
+ */
+trait ExhaustiveCheck[A] extends Serializable { self =>
+  def allValues: Stream[A]
+
+  def map[B](f: A => B): ExhaustiveCheck[B] = new ExhaustiveCheck[B] {
```

Comment:
This isn’t safe. I think it should be imap to preserve the exhaustivity.

Reply:
Thanks, good catch!